src/nbl/asset/utils/CPolygonGeometryManipulator.cpp

```diff
@@ -135,6 +136,86 @@ core::smart_refctd_ptr<ICPUPolygonGeometry> CPolygonGeometryManipulator::createU
     return outGeometry;
 }
 
+
+namespace
+{
+  bool isAttributeValEqual(const ICPUPolygonGeometry::SDataView& view, uint64_t index1, uint64_t index2, float epsilon)
+  {
+    if (!view) return true;
+    const auto channelCount = getFormatChannelCount(view.composed.format);
+    switch (view.composed.rangeFormat)
+    {
+      case IGeometryBase::EAABBFormat::U64:
+      case IGeometryBase::EAABBFormat::U32:
+      {
+        hlsl::uint64_t4 val1, val2;
+        view.decodeElement<hlsl::uint64_t4>(index1, val1);
+        view.decodeElement<hlsl::uint64_t4>(index2, val2);
+        for (auto channel_i = 0u; channel_i < channelCount; channel_i++)
+          if (val1[channel_i] != val2[channel_i]) return false;
+        break;
+      }
+      case IGeometryBase::EAABBFormat::S64:
+      case IGeometryBase::EAABBFormat::S32:
+      {
+        hlsl::int64_t4 val1, val2;
+        view.decodeElement<hlsl::int64_t4>(index1, val1);
+        view.decodeElement<hlsl::int64_t4>(index2, val2);
+        for (auto channel_i = 0u; channel_i < channelCount; channel_i++)
+          if (val1[channel_i] != val2[channel_i]) return false;
+        break;
+      }
+      default:
+      {
+        hlsl::float64_t4 val1, val2;
+        view.decodeElement<hlsl::float64_t4>(index1, val1);
+        view.decodeElement<hlsl::float64_t4>(index2, val2);
+        for (auto channel_i = 0u; channel_i < channelCount; channel_i++)
+        {
+          const auto diff = abs(val1[channel_i] - val2[channel_i]);
+          if (diff > epsilon) return false;
+        }
+        break;
+      }
+    }
+    return true;
+  }
+
+  bool isAttributeDirEqual(const ICPUPolygonGeometry::SDataView& view, uint64_t index1, uint64_t index2, float epsilon)
+{
+  if (!view) return true;
+  const auto channelCount = getFormatChannelCount(view.composed.format);
+  switch (view.composed.rangeFormat)
+  {
+    case IGeometryBase::EAABBFormat::U64:
+    case IGeometryBase::EAABBFormat::U32:
+    {
+      hlsl::uint64_t4 val1, val2;
+      view.decodeElement<hlsl::uint64_t4>(index1, val1);
+      view.decodeElement<hlsl::uint64_t4>(index2, val2);
+      return (1.0 - hlsl::dot(val1, val2)) < epsilon;
+    }
+    case IGeometryBase::EAABBFormat::S64:
+    case IGeometryBase::EAABBFormat::S32:
+    {
+      hlsl::int64_t4 val1, val2;
+      view.decodeElement<hlsl::int64_t4>(index1, val1);
+      view.decodeElement<hlsl::int64_t4>(index2, val2);
+      return (1.0 - hlsl::dot(val1, val2)) < epsilon;
+    }
+    default:
+    {
+      hlsl::float64_t4 val1, val2;
+      view.decodeElement<hlsl::float64_t4>(index1, val1);
+      view.decodeElement<hlsl::float64_t4>(index2, val2);
+      return (1.0 - hlsl::dot(val1, val2)) < epsilon;
```

Comment:
only do this if the component count of the format is 3, otherwise its an equality comparison (because if someone uses 2, they are probably using spherical mappings, theta/phi angles or something else we don't have the semantics for)